### PR TITLE
DAOS-6273 control: fix storage query usage for scm emulated with ram

### PR DESCRIPTION
**Pull request by tanabarr**
Fix issue that when using a ram SCM entry (tmpfs) issuing a dmg
storage query usage command will return an error and not reflect the
correct amount of storage.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Comment by tanabarr** [comment]
example output from storage query usage when using RAMdisk to emulate SCM:
```
tmpfs on /mnt/daos1 type tmpfs (rw,noatime,size=12582912k,gid=989)
tmpfs on /mnt/daos0 type tmpfs (rw,noatime,size=12582912k,gid=989)
bash-4.2$ dmg storage query usage
Hosts   SCM-Total SCM-Free SCM-Used NVMe-Total NVMe-Free NVMe-Used
-----   --------- -------- -------- ---------- --------- ---------
wolf-71 26 GB     26 GB    1 %      1.5 TB     1.5 TB    0 %
wolf-72 26 GB     26 GB    1 %      1.5 TB     1.5 TB    0 %
bash-4.2$ dmg pool create -z 100G
Creating DAOS pool with automatic storage allocation: 100 GB NVMe + 6.00% SCM
Pool created with 6.00% SCM/NVMe ratio
---------------------------------------
  UUID          : be22f391-8eeb-45ae-ba48-23b604e9fdce
  Service Ranks : [1-3]
  Storage Ranks : [0-3]
  Total Size    : 106 GB
  SCM           : 6.0 GB (1.5 GB / rank)
  NVMe          : 100 GB (25 GB / rank)

bash-4.2$ dmg storage query usage
Hosts   SCM-Total SCM-Free SCM-Used NVMe-Total NVMe-Free NVMe-Used
-----   --------- -------- -------- ---------- --------- ---------
wolf-71 26 GB     22 GB    13 %     1.5 TB     1.4 TB    4 %
wolf-72 26 GB     22 GB    13 %     1.5 TB     1.4 TB    4 %
bash-4.2$ dmg pool create -z 400G
Creating DAOS pool with automatic storage allocation: 400 GB NVMe + 6.00% SCM
Pool created with 6.00% SCM/NVMe ratio
---------------------------------------
  UUID          : ba238b5c-80a5-4831-a620-f309a507585e
  Service Ranks : [1-3]
  Storage Ranks : [0-3]
  Total Size    : 424 GB
  SCM           : 24 GB (6.0 GB / rank)
  NVMe          : 400 GB (100 GB / rank)

bash-4.2$ dmg storage query usage
Hosts   SCM-Total SCM-Free SCM-Used NVMe-Total NVMe-Free NVMe-Used
-----   --------- -------- -------- ---------- --------- ---------
wolf-71 26 GB     10 GB    60 %     1.5 TB     1.2 TB    18 %
wolf-72 26 GB     10 GB    61 %     1.5 TB     1.2 TB    18 %
bash-4.2$
```

File: src/control/server/ctl_storage_rpc_test.go [hunk 1]
```diff
@@ -575,7 +594,10 @@ func TestServer_CtlSvc_StorageScan_PostIOStart(t *testing.T) {
 			},
 		},
 		"scan bdev meta with multiple io servers up with multiple nvme namespaces": {
-			req: &StorageScanReq{Nvme: &ScanNvmeReq{Meta: true}},
+			req: &StorageScanReq{
+				Scm:  new(ScanScmReq),
```
Comment (kjacque) [inline src/control/server/ctl_storage_rpc_test.go:598]: Do we still behave correctly if these are nil? since they're pointers

**Review by kjacque — APPROVED** [review]
Overall LGTM.

File: src/control/server/ctl_storage_rpc_test.go (hunk 1 above)
Comment (tanabarr) [reply src/control/server/ctl_storage_rpc_test.go:598]: The sub functions will refuse nil, the references are always non-nil as they get populated in the control API function that calls StorageScan (StorageScan handler is only ever called through the control API).

Comment (kjacque) [reply src/control/server/ctl_storage_rpc_test.go:598]: Cool, got it.

**Review by mjmac — APPROVED** [review]

**Pull request merged**